feature-libs/cart/import-export/core/services/import-csv.service.ts

```diff
@@ -49,12 +36,59 @@ export class ImportCsvService {
   }
 
   loadCsvData(file: File): Observable<string[][]> {
-    return this.loadFile(file).pipe(
-      map((res) => this.readCsvData(res as string))
-    );
+    return this.importService
+      .loadTextFile(file)
+      .pipe(map((res) => this.readCsvData(res as string)));
+  }
+
+  isReadableFile(
+    isDataParsable?: (data: string[][]) => boolean
+  ): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+      const errors: ValidationErrors = {};
+      const file: File = control.value[0];
+      return (this.importService.loadTextFile(file) as Observable<string>).pipe(
+        tap((data: string) => {
+          this.validEmpty(data, errors);
+        }),
+        map((res) => this.readCsvData(res)),
+        tap((data: string[][]) => {
+          this.validTooManyEntries(data, errors);
+          this.validNotParsable(data, errors, isDataParsable);
+        }),
+        map(() => (Object.keys(errors).length === 0 ? null : errors))
+      );
+    };
   }
 
-  private get separator() {
+  protected validEmpty(data: string, errors: ValidationErrors) {
+    if (data.toString().length === 0) {
+      errors.empty = true;
+    }
+  }
+
+  protected validTooManyEntries(data: string[][], errors: ValidationErrors) {
+    if (this.maxEntries && data.length > this.maxEntries) {
+      errors.tooManyEntries = { maxEntries: this.maxEntries };
+    }
+  }
+
+  protected validNotParsable(
```

Comment:
nitpick: let's stick to the convention: method names start with verbs. What about `validate...`?